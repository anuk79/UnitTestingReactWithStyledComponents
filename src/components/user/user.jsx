// @flow

// vendors
import React from 'react';

const UserView = (props) => {
    return (
        <React.Fragment>
            <h3>{'User Details:'}</h3>
            <div>
                <label>{'First Name: '}</label>
                <span>{props.userDetails.firstName}</span>
            </div>
            <div>
                <label>{'Last Name: '}</label>
                <span>{props.userDetails.lastName}</span>
            </div>
        </React.Fragment>
    );
}

export default UserView;