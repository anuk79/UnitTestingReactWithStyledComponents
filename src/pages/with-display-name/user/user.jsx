// @flow

// vendors
import React, { Component } from 'react';

// components
import UserView from '../../../components/user/user';

// styles
import {
    LoadingSpinnerStyled,
    ErrorStyled,
} from './user.styled';

export class User extends Component {
    render() {
        const { fetching, userDetails, errorFlag } = this.props;
        let content;
        if(fetching) {
            content = <LoadingSpinnerStyled customClassName="user-loading-spinner" />
        } else if(errorFlag) {
            content = (
                <ErrorStyled>{'Some error occurred. Please try again later.'}</ErrorStyled>
            );
        } else {
            content = <UserView userDetails={userDetails} />
        }
        return content;
    }
}

export default User;