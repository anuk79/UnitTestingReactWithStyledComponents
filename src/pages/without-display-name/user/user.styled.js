// vendors
import styled from 'styled-components';

// components
import LoadingSpinner from '../../../components/loading-spinner/loading-spinner';

export const LoadingSpinnerStyled = styled(LoadingSpinner)`
    margin: 20px;
`;

export const ErrorStyled = styled.div`
    color: red;
    font-size: 14px;
    margin: 20px;
`;