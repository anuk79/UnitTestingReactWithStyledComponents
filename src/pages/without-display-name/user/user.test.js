
// vendors
import React from "react";
import { shallow } from 'enzyme';
import { shallowToJson } from "enzyme-to-json";

// components
import { User } from './user';

describe("User component", () => {
    it("should render correctly when data is fetching", () => {
        const wrapper = shallow(<User fetching={true} />);

        expect(shallowToJson(wrapper)).toMatchSnapshot();

        expect(wrapper.find('Styled(LoadingSpinner)').length).toBe(1);
    });

    it("should render correctly when some error", () => {
        const wrapper = shallow(<User errorFlag={true} />);

        expect(shallowToJson(wrapper)).toMatchSnapshot();

        expect(wrapper.find('styled.div').length).toBe(1);
    });

    it("should render correctly when no error and data fetching is done", () => {
        const wrapper = shallow(<User userDetails={{}} />);

        expect(shallowToJson(wrapper)).toMatchSnapshot();
        
        expect(wrapper.find('UserView').length).toBe(1);
        expect(wrapper.find('UserView').prop('userDetails')).toEqual({});
    });
});