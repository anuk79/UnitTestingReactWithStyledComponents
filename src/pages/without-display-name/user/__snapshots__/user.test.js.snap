// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User component should render correctly when data is fetching 1`] = `
<Styled(LoadingSpinner)
  customClassName="user-loading-spinner"
/>
`;

exports[`User component should render correctly when some error 1`] = `
<styled.div>
  Some error occurred. Please try again later.
</styled.div>
`;

exports[`User component should render correctly when no error and data fetching is done 1`] = `
<UserView
  userDetails={Object {}}
/>
`;
